src/GraphQL/Validation/Complexity/FieldImpactContext.cs

```diff
@@ -0,0 +1,80 @@
+using GraphQL.Execution;
+using GraphQL.Types;
+using GraphQL.Validation.Rules.Custom;
+using GraphQLParser.AST;
+
+namespace GraphQL.Validation.Complexity;
+
+/// <summary>
+/// Provides contextual information about the field being analyzed.
+/// </summary>
+public struct FieldImpactContext
```

Comment:
Using a struct here with limited amount of internal fields to speed complexity analysis and minimize allocations, as this is constructed for every analyzed field.  Using a struct means that it cannot be derived from, but additional fields can be added to a derived `ComplexityConfiguration` instance, which is accessible from this struct.